src/FSharp.Analyzers.Cli/Program.fs

```diff
@@ -0,0 +1,186 @@
+﻿open System
+open System.IO
+open FSharp.Compiler.SourceCodeServices
+open FSharp.Compiler.Text
+open ProjectSystem
+open Argu
+open FSharp.Analyzers.SDK
+open GlobExpressions
+
+type Arguments =
+    | Project of string
+    | Analyzer of string
+    | Fail_On_Warnings of string list
+    | Ignore_Files of string list
+    | Verbose
+with
+    interface IArgParserTemplate with
+        member s.Usage = ""
+
+let mutable verbose = false
+
+let checker =
+    FSharpChecker.Create(
+        projectCacheSize = 200,
+        keepAllBackgroundResolutions = true,
+        keepAssemblyContents = true,
+        ImplicitlyStartBackgroundWork = true)
+
+let projectSystem = ProjectController(checker)
+let parser = ArgumentParser.Create<Arguments>()
+
+let rec mkKn (ty: System.Type) =
+    if Reflection.FSharpType.IsFunction(ty) then
+        let _, ran = Reflection.FSharpType.GetFunctionElements(ty)
+        let f = mkKn ran
+        Reflection.FSharpValue.MakeFunction(ty, fun _ -> f)
+    else
+        box ()
+
+
+let printInfo (fmt: Printf.TextWriterFormat<'a>) : 'a =
+    if verbose then
+        Console.ForegroundColor <- ConsoleColor.DarkGray
+        printf "Info : "
+        Console.ForegroundColor <- ConsoleColor.White
+        printfn fmt
+    else
+        unbox (mkKn typeof<'a>)
+
+let printError text arg =
+    Console.ForegroundColor <- ConsoleColor.Red
+    printf "Error : "
+    printfn text arg
+    Console.ForegroundColor <- ConsoleColor.White
+
+let loadProject file =
+    async {
+        let! projLoading = projectSystem.LoadProject file ignore FSIRefs.TFM.NetCore (fun _ _ _ -> ())
+        let filesToCheck =
+            match projLoading with
+            | ProjectResponse.Project proj ->
+                printInfo "Project %s loaded" file
+                proj.projectFiles
+                |> List.choose (fun file ->
+                    projectSystem.GetProjectOptions file
+                    |> Option.map (fun opts -> file, opts)
+                )
+            | ProjectResponse.ProjectError(errorDetails) ->
+                printError "Project loading faield: %A" errorDetails
+                []
```

Comment:
Do you think the tool should exit with a non-zero exit code if we can't load the project?

Reply:
Also typo `faield`